internal/volume/volumes_information.go

```diff
@@ -551,6 +551,22 @@ func (v *BackupVolumesInformation) generateVolumeInfoFromPVB() {
 	v.volumeInfos = append(v.volumeInfos, tmpVolumeInfos...)
 }
 
+func (v *BackupVolumesInformation) getVolumeSnapshotClasses(
+	classes *snapshotv1api.VolumeSnapshotClassList,
+) (*snapshotv1api.VolumeSnapshotClassList, error) {
+	if classes != nil {
+		return classes, nil
+	}
+
+	vsClassList := new(snapshotv1api.VolumeSnapshotClassList)
+	if err := v.crClient.List(context.TODO(), vsClassList); err != nil {
+		v.logger.WithError(err).Errorf("cannot list VolumeSnapshotClass %s", err.Error())
```

Comment:
If we generate an error log here, the backup will be partiallyFailed. So probably we need to check `apierrors.IsNotFound`, if it is true, no log should be printed, otherwise, I think a warning log is enough.

Reply:
Because this function is only called when the DataUpload CR exists, we should print some information on failing to get the VolumeSnapshotClasses list, no matter the reason for the error.

This condition check already can avoid the scenario described in issue #7823.

As a result, we should print a log error without checking the error type, but it's reasonable to lower the log level to a warning.

Reply:
VolumeSnapshotClass is only used by data mover backup, for the restore, it can be absent.